src/tests/test_fd_util.h

```diff
@@ -77,4 +77,80 @@ class AlgorithmTest : public ::testing::Test {
              {tests::kIowa1kk, 28573},
              {tests::kLegacyPayors, 43612}}};
 };
+
+template <typename T>
+class ApproximateFDTest : public ::testing::Test {
+protected:
+    // alias for choosing datasets' hashes specialization
+    using AlgorithmType = T;
+
+    static std::unique_ptr<algos::FDAlgorithm> CreateAndConfToLoad(CSVConfig const& csv_config) {
+        using namespace config::names;
+        using algos::ConfigureFromMap, algos::StdParamsMap;
+
+        std::unique_ptr<algos::FDAlgorithm> algorithm = std::make_unique<T>();
+        ConfigureFromMap(*algorithm, StdParamsMap{{kTable, MakeInputTable(csv_config)}});
+        return algorithm;
+    }
+
+    static algos::StdParamsMap GetParamMap(CSVConfig const& csv_config) {
+        using namespace config::names;
+        return {
+                {kCsvConfig, csv_config},
+                {kCustomRandom, std::make_pair(true, 47)},
+        };
+    }
+
+    static void PerformConsistentHashTestOn(std::vector<CSVConfigHash> const& config_hashes) {
+        try {
+            for (auto const& [csv_config, hash] : config_hashes) {
+                auto algorithm = CreateAlgorithmInstance(csv_config);
+                algorithm->Execute();
+                EXPECT_EQ(algorithm->Fletcher16(), hash)
+                        << "FD collection hash changed for " << csv_config.path.filename();
+            }
+        } catch (std::runtime_error& e) {
+            std::cout << "Exception raised in test: " << e.what() << std::endl;
+            FAIL();
+        }
+        SUCCEED();
+    }
+
+public:
+    static std::unique_ptr<algos::FDAlgorithm> CreateAlgorithmInstance(CSVConfig const& config) {
+        return algos::CreateAndLoadAlgorithm<T>(GetParamMap(config));
+    }
+};
+
+// testing data
+template <typename Alg>
+struct ApproximateDatasets {
+    inline static std::vector<CSVConfigHash> const kLightDatasets;
+    inline static std::vector<CSVConfigHash> const kHeavyDatasets;
+};
+
+// specialization fd for EulerFD
+template <>
+struct ApproximateDatasets<algos::EulerFD> {
+    inline static std::vector<CSVConfigHash> const kLightDatasets = {{
+            {tests::kCIPublicHighway10k, 33398},
+            {tests::kNeighbors10k, 43368},
+            {tests::kWdcAstronomical, 2902},  // answer is 9 / 15
+            {tests::kWdcAppearances, 64338},  // answer is 3 / 4
+            {tests::kWdcAstrology, 40815},    // answer is 34 / 20
+            {tests::kWdcSymbols, 28289},
+            {tests::kBreastCancer, 15121},
+            {tests::kWdcKepler, 17294},  // empty answer, 0 clusters after stripping
+    }};
+
+    inline static std::vector<CSVConfigHash> const kHeavyDatasets = {{
+            {tests::kAdult, 23075},
+            {tests::kCIPublicHighway, 13035},
+            {tests::kEpicMeds, 26201},  // answer is 15 / 16
+            {tests::kEpicVitals, 2083},
+            {tests::kIowa1kk, 57837},  // answer is 2531 / 1584 (average 2k, it is bad seed :(
```

Comment:
Correct brackets. =)

Reply:
How? Like in datasets for not approximate algorithms?
`
... = {
{{..., ...},
 {..., ...}}};
`
Like this?